### DISCUSSION TITLE
lesson10 [FAIL. Reason: Invalid data] testBobBalance()

### DISCUSSION THREAD

**Discussion by RaferGuo** [Cyfrin/foundry-full-course-cu#1476] [Q&A] [answered]
i dont know how to debug it， thx
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/95877aef-7e34-49ff-9cb5-2381c17779de)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/1fa2ebfc-501c-4c31-8778-b9aedb2ac212)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/6b325d26-8b09-4ec7-8d67-58ddd7ba9cb9)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/e4f45eab-0432-4ac5-87b4-b3ab9d18c0d0)

**Comment by EngrPips** [comment]
Hello @RaferGuo , i have been starring at your code for about 10 minutes now and i can't find where you got it wrong, can you maybe push the code to github and share me the link to the repo so that i clone it, set it up locally on my machine and try debug it.

**Reply by RaferGuo** [reply]
here is the github link:https://github.com/RaferGuo/foundry-erc20-23
by the way when i test AllowancesWorks,i meet the same problem
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/1a6fc49e-5244-493e-b2f7-7cd41bc76aca)

**Comment by EngrPips** [comment, marked as answer]
Hello @RaferGuo , i would suggest you run `foundryUp` and then re-run the test. I set up the code on my machine and the test run just fine, picture below 

<img width="1440" alt="Screenshot 2024-03-13 at 11 31 28 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/7e8d096c-36d1-4033-a1e6-b4ee6422829f">

run `foundryUp` and try again and let me kmow if it works.

**Reply by RaferGuo** [reply]
yes，run foundryup and then it works. amazing，but still dont know the reason.thx

**Reply by EngrPips** [reply]
Glad it worked, your foundry needed an upgrade thats why it was malfunctioning.
